sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportEventHubClient.cs

```diff
@@ -45,10 +53,12 @@ internal abstract class TransportEventHubClient
         /// </summary>
         ///
         /// <param name="producerOptions">The set of options to apply when creating the producer.</param>
+        /// <param name="defaultRetryPolicy">The default retry policy to use if no retry options were specified in the <paramref name="producerOptions" />.</param>
         ///
         /// <returns>An Event Hub producer configured in the requested manner.</returns>
         ///
-        public abstract EventHubProducer CreateProducer(EventHubProducerOptions producerOptions);
+        public abstract EventHubProducer CreateProducer(EventHubProducerOptions producerOptions,
+                                                        EventHubRetryPolicy defaultRetryPolicy);
```

Comment:
The factories are only used in the case where we can't guarantee that the retry policy is stable between when the containing class is created and the inner client is lazily created.   We're able to get a stable snapshot here since there's no lazy creation.